Description: Fix 32 bits ARM & MIPS build
Author: Jean Bpatiste Favre <debian@jbfavre.org>
Origin: other
Bug-Upstream: https://github.com/apache/trafficserver/issues/4361
Reviewed-by: Aron Xu <aron@debian.org>
Last-Update: 2018-10-17
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/include/tscore/ink_atomic.h
+++ b/include/tscore/ink_atomic.h
@@ -93,7 +93,7 @@ extern ink_mutex __global_death;
 
 template <>
 inline int64_t
-ink_atomic_swap<int64_t>(pvint64 mem, int64_t value)
+ink_atomic_swap<int64_t>(int64_t *mem, int64_t value)
 {
   int64_t old;
   ink_mutex_acquire(&__global_death);
@@ -105,7 +105,7 @@ ink_atomic_swap<int64_t>(pvint64 mem, in
 
 template <>
 inline bool
-ink_atomic_cas<int64_t>(pvint64 mem, int64_t old, int64_t new_value)
+ink_atomic_cas<int64_t>(int64_t *mem, int64_t old, int64_t new_value)
 {
   int64_t curr;
   ink_mutex_acquire(&__global_death);
@@ -120,7 +120,7 @@ ink_atomic_cas<int64_t>(pvint64 mem, int
 
 template <typename Amount>
 static inline int64_t
-ink_atomic_increment(pvint64 mem, Amount value)
+ink_atomic_increment(int64_t *mem, Amount value)
 {
   int64_t curr;
   ink_mutex_acquire(&__global_death);
@@ -132,7 +132,7 @@ ink_atomic_increment(pvint64 mem, Amount
 
 template <typename Amount>
 static inline int64_t
-ink_atomic_decrement(pvint64 mem, Amount value)
+ink_atomic_decrement(int64_t *mem, Amount value)
 {
   int64_t curr;
   ink_mutex_acquire(&__global_death);
@@ -144,7 +144,7 @@ ink_atomic_decrement(pvint64 mem, Amount
 
 template <typename Amount>
 static inline uint64_t
-ink_atomic_increment(pvuint64 mem, Amount value)
+ink_atomic_increment(uint64_t *mem, Amount value)
 {
   uint64_t curr;
   ink_mutex_acquire(&__global_death);
@@ -156,7 +156,7 @@ ink_atomic_increment(pvuint64 mem, Amoun
 
 template <typename Amount>
 static inline uint64_t
-ink_atomic_decrement(pvuint64 mem, Amount value)
+ink_atomic_decrement(uint64_t *mem, Amount value)
 {
   uint64_t curr;
   ink_mutex_acquire(&__global_death);
--- a/src/traffic_cache_tool/Makefile.inc
+++ b/src/traffic_cache_tool/Makefile.inc
@@ -42,6 +42,7 @@ traffic_cache_tool_traffic_cache_tool_LD
     $(top_builddir)/src/tscore/.libs/ink_error.o \
     $(top_builddir)/src/tscore/.libs/ink_file.o \
     $(top_builddir)/src/tscore/.libs/ink_memory.o \
+    $(top_builddir)/src/tscore/.libs/ink_mutex.o \
     $(top_builddir)/src/tscore/.libs/ink_string.o \
     $(top_builddir)/src/tscore/.libs/BufferWriterFormat.o \
     $(top_builddir)/src/tscpp/util/.libs/TextView.o \
